src/plugins/discover/public/application/utils/state_management/discover_slice.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Filter, Query } from '../../../../../data/public';
+import { DiscoverServices } from '../../../build_services';
+import { RootState } from '../../../../../data_explorer/public';
+
+export interface DiscoverState {
+  /**
+   * Columns displayed in the table
+   */
+  columns?: string[];
+  /**
+   * Array of applied filters
+   */
+  filters?: Filter[];
+  /**
+   * Used interval of the histogram
+   */
+  interval?: string;
```

Comment:
Can't these just be retrieved from the global state, because they're universal and not specific to discover?

Reply:
I copied this over form the legacy discover state. When Anan adds the doc table, she will update this to be a more accurate representation of the discover state.